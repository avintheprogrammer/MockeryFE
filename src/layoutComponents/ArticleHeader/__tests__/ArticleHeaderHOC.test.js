import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { ArticleHeader } from '../index';
import Header from '../../../components/Article/ArticleHeader';
import mock from '../__mocks__/ArticleHeaderHOC.mock';

configure({ adapter: new Adapter() });

test('Article Header renders correctly', () => {
  const tree = shallow(<ArticleHeader data={mock} />);
  expect(toJson(tree)).toMatchSnapshot();
});

test('Render the article header if data is valid.', () => {
  const wrapper = shallow(<ArticleHeader data={mock} dataId={0} />);

  expect(
    wrapper.contains(
      <Header
        data={mock}
        hasBackground={false}
        isTransporter={false}
        isSpecialReport={false}
        dataId={0}
        brand=""
      />,
    ),
  ).toBe(true);
});
