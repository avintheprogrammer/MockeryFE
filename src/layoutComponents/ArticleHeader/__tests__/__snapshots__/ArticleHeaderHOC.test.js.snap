// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Header renders correctly 1`] = `
<div
  id={0}
>
  <Connect(ArticleHeader)
    brand=""
    data={
      Object {
        "article": Object {
          "featuredMedia": null,
          "native": false,
          "section": null,
          "title": "hello",
        },
      }
    }
    dataId={0}
    hasBackground={false}
    isSpecialReport={false}
    isTransporter={false}
  />
</div>
`;
